### ISSUE TITLE
Script Event is sending continuous event after blocking it

### ISSUE THREAD

**Issue by cojajes221** [Yimura/YimMenu#145]
**Describe the bug**
Script Event is still in queue even after blocking it which will result into getting all the script events sended at once, once unloaded
Its also a related issue to #86 

**To Reproduce**
1. Block an script event with scripted_game_event
2. Unload the menu
3. Get the whole power of script events in your face

**Expected behavior**
Normal behaviour after unload

** Suggested solution **
An m_send_event_ack to acknowledge the script event and get rid of the "queue" in received_event for the rockstar
event SCRIPTED_GAME_EVENT
An example here: https://www.unknowncheats.me/forum/2996769-post24.html

**Build**
Select which applies for you:
- [x] I used YimMenu unmodified
- [ ] I use my own version of YimMenu

**Comment by ImpenetrableSystem** [comment]
Can confirm it, I could look later into it however from the first look the handler in ida looks good for the example given here.
@Yimura could you also take a look over it?

**Comment by Yimura** [comment]
Alright I'll take a look at it, which script event exactly are you talking about?

My assumption was that returning true in the script event handler would remove items from the queue, is this assumption wrong?
To my knowledge there's only one script event which keeps getting queue'd over and over is the Bounty event.

As a possible solution I can move the script event detection to the [received event hook](https://github.com/Yimura/YimMenu/blob/master/BigBaseV2/src/hooks/received_event.cpp) which is the same hook as you linked in to.

**Comment by ImpenetrableSystem** [comment]
Yeah thats the problem ever single script event I tested which gets blocked by return true is getting into this queue and in order to send this m_send_event_ack in order to acknowledge the script event and in the same time blocking it. So basicly the best solution I got is:

`
case SCRIPTED_GAME_EVENT:
	{
		auto game_event = CScriptedGameEvent();
		buffer->ReadDword(&game_event.m_args_size, 32);
		if (game_event.m_args_size <= 0x1AF)
			buffer->ReadArray(&game_event.m_args, 8 * game_event.m_args_size);
		if (hooking::scripted_game_event(&game_event, source_player) == false)
			{
				g_pointers->m_send_event_ack(event_manager, source_player, target_player, event_index, event_handled_bitset);
				return false;
			}
		buffer->Seek(0);
		return true;
	}
`
^^ Untested

As from testing it doesnt matter which script event is sended every blocked one is getting into this queue.
If you want look into it: CScriptedGameEvent vtable with ClassInformer and then 6th method is the handler for this however dont mix up with 7th method which is scripted_game_event
